icu4c/source/test/intltest/numbertest_parse.cpp

```diff
@@ -408,5 +409,83 @@ void NumberParserTest::testInfiniteRecursion() {
+    IcuTestErrorCode status(*this, "test13733_StrictAndLenient");
+
+    static const struct TestCase {
+        const char16_t* inputString;
+        const char16_t* patternString;
+        long expectedStrictParse;
+        long expectedLenientParse;
+    } cases[] = { {u"CA$ 12", u"¤ 0", 12, 12},
+                  {u"CA$12", u"¤0", 12, 12},
+                  {u"CAD 12", u"¤¤ 0", 12, 12},
+                  {u"12 CAD", u"0 ¤¤", 12, 12},
+                  {u"12 Canadian dollars", u"0 ¤¤¤", 12, 12},
+                  {u"$12 ", u"¤¤¤¤0", 12, 12},
+                  {u"12$", u"0¤¤¤¤", 12, 12},
+                  {u"CA$ 12", u"¤0", NULL, 12},
+                  {u"CA$ 12", u"0 ¤¤", NULL, 12},
+                  {u"CA$ 12", u"0 ¤¤¤", NULL, 12},
+                  {u"CA$ 12", u"¤¤¤¤0", NULL, 12},
+                  {u"CA$ 12", u"0¤¤¤¤", NULL, 12},
+                  {u"CA$12", u"¤ 0", NULL, 12},
+                  {u"CA$12", u"¤¤ 0", NULL, 12},
+                  {u"CA$12", u"0 ¤¤", NULL, 12},
+                  {u"CA$12", u"0 ¤¤¤", NULL, 12},
+                  {u"CA$12", u"0¤¤¤¤", NULL, 12},
+                  {u"CAD 12", u"¤0", NULL, 12},
+                  {u"CAD 12", u"0 ¤¤", NULL, 12},
+                  {u"CAD 12", u"0 ¤¤¤", NULL, 12},
+                  {u"CAD 12", u"¤¤¤¤0", NULL, 12},
+                  {u"CAD 12", u"0¤¤¤¤", NULL, 12},
+                  {u"12 CAD", u"¤ 0", NULL, 12},
+                  {u"12 CAD", u"¤0", NULL, 12},
+                  {u"12 CAD", u"¤¤ 0", NULL, 12},
+                  {u"12 CAD", u"¤¤¤¤0", NULL, 12},
+                  {u"12 CAD", u"0¤¤¤¤", NULL, 12},
+                  {u"12 Canadian dollars", u"¤ 0", NULL, 12},
+                  {u"12 Canadian dollars", u"¤0", NULL, 12},
+                  {u"12 Canadian dollars", u"¤¤ 0", NULL, 12},
+                  {u"12 Canadian dollars", u"¤¤¤¤0", NULL, 12},
+                  {u"12 Canadian dollars", u"0¤¤¤¤", NULL, 12},
+                  {u"$12 ", u"¤ 0", NULL, 12},
+                  {u"$12 ", u"¤¤ 0", NULL, 12},
+                  {u"$12 ", u"0 ¤¤", NULL, 12},
+                  {u"$12 ", u"0 ¤¤¤", NULL, 12},
+                  {u"$12 ", u"0¤¤¤¤", NULL, 12},
+                  {u"12$", u"¤ 0", NULL, 12},
+                  {u"12$", u"¤0", NULL, 12},
+                  {u"12$", u"¤¤ 0", NULL, 12},
+                  {u"12$", u"0 ¤¤", NULL, 12},
+                  {u"12$", u"0 ¤¤¤", NULL, 12},
+                  {u"12$", u"¤¤¤¤0", NULL, 12} };
+    for (auto& cas : cases) {
+        UnicodeString inputString(cas.inputString);
+        UnicodeString patternString(cas.patternString);
+        long parsedStrictValue = NULL;
+        long parsedLenientValue = NULL;
+        ParsePosition ppos;
+
+        DecimalFormatSymbols dfs(Locale::getEnglish(), status);
+        DecimalFormat df(patternString, dfs, status);
+        df.setLenient(FALSE);
+        CurrencyAmount* ca_strict = df.parseCurrency(inputString, ppos);
```

Comment:
Do you need to clean up the CurrencyAmount?  If so, please use LocalPointer.  If not, please add a comment.

Reply:
Changed to <LocalPointer>